pydantic/config.py

```diff
@@ -1,17 +1,19 @@
 from __future__ import annotations as _annotations
 
-from enum import Enum
 from typing import Callable
 
 from typing_extensions import Literal, TypedDict
 
 __all__ = 'ConfigDict', 'Extra'
 
 
-class Extra(str, Enum):
-    allow = 'allow'
-    ignore = 'ignore'
-    forbid = 'forbid'
+class Extra:
+    allow: Literal['allow'] = 'allow'
+    ignore: Literal['ignore'] = 'ignore'
+    forbid: Literal['forbid'] = 'forbid'
```

Comment:
We could easily add a deprecation warning on attribute access

Reply:
Do we need to change this? I'm thinking we can just keep it as the enum for now and still add the deprecation warning on access. Just seems that much less likely to break things. I would give it a nonzero chance that someone is relying on the Enum-ness of it somewhere (e.g., using it as an argument to a FastAPI endpoint or something).

Reply:
I guess if it were up to me, I would be more in favor of keeping it as an Enum or dropping it entirely than adding this, which I think as a user I would find a bit more surprising; admittedly it will keep more cases working without issue, but if it does cause errors, they will probably be harder/more surprising to debug than if the class was just missing.

If you can get it to say "this class is deprecated" when accessing the class attributes I think that's fine.

Reply:
Yup that’s what I was planning on doing.